process_report/invoices/pi_specific_invoice.py

```diff
@@ -92,20 +92,44 @@ def _get_pi_dataframe(self, data, pi):
             if column_name in pi_projects.columns:
                 column_sums.append(pi_projects[column_name].sum())
                 sum_columns_list.append(column_name)
-        pi_projects.loc[len(pi_projects)] = (
-            None  # Adds a new row to end of dataframe initialized with None
+
+        # Create a new row with proper dtypes
+        new_row = {col: None for col in pi_projects.columns}
```

Comment:
Within this repo and many of our other ones, our team generally only writes comments when the rationale for a piece of code is not immediately obvious. This is usually when a piece of code is quite complex, structured weirdly, or its purpose not clear. In which case, the comment would explain WHY the code is there.

If the code is self-explanatory, in the case of this line, it's not helpful to have a comment explaining what the code does. I would suggest you remove this comment.